test/base/src/androidTest/kotlin/io/realm/test/shared/EmbeddedObjectTests.kt

```diff
@@ -0,0 +1,1161 @@
+            assertEquals("Initial child", id)
+        }
+    }
+
+    @Test
+    fun setWillDeleteEmbeddedObject() {
+        val parent = realm.writeBlocking {
+            copyToRealm(EmbeddedParent().apply { child = EmbeddedChild() })
+        }
+        realm.query<EmbeddedChild>().find().single()
+        realm.writeBlocking {
+            findLatest(parent)!!.apply {
+                child = null
+            }
+        }
+        realm.query<EmbeddedChild>().find().none()
+    }
+
+    // @org.junit.Test
+    // fun dynamicRealm_settingParentFieldDeletesChild() =
+    //     DynamicRealm.getInstance(realm.configuration).use { realm ->
+    //         realm.executeTransaction {
+    //             val parent = realm.createObject("EmbeddedSimpleParent", "parent")
+    //             val child = realm.createEmbeddedObject("EmbeddedSimpleChild", parent, "child")
+    //
+    //             Assert.assertEquals(1, realm.where("EmbeddedSimpleChild").count())
+    //             parent.setObject("child", null)
+    //             Assert.assertFalse(child.isValid)
+    //             Assert.assertEquals(0, realm.where("EmbeddedSimpleChild").count())
+    //         }
+    //     }
+    //
+
+    @Test
+    fun set_unmanaged() {
+        val parent = realm.writeBlocking {
+            copyToRealm(EmbeddedParent())
+        }
+        realm.query<EmbeddedChild>().find().none()
+        realm.writeBlocking {
+            findLatest(parent)!!.apply {
+                child = EmbeddedChild().apply {
+                    id = "child1"
+                    innerChild = EmbeddedInnerChild()
+                }
+            }
+        }
+        realm.query<EmbeddedChild>().find().single().run {
+            assertEquals("child1", id)
+        }
+        realm.query<EmbeddedChild>().find().single()
+    }
+
+    @Test
+    fun set_managed() {
+        realm.writeBlocking {
+            val parent1 = copyToRealm(EmbeddedParent().apply { id = "parent1" })
+            val parent2 = copyToRealm(EmbeddedParent().apply { id = "parent2" })
+
+            parent1.child = EmbeddedChild("child1")
+            parent2.child = parent1.child
+        }
+        val children = realm.query<EmbeddedChild>().find()
+        children.run {
+            assertEquals(2, size)
+            forEach { assertEquals("child1", it.id) }
+        }
+    }
+
+    @Test
+    fun list_add() {
+        realm.writeBlocking {
+            val parent = copyToRealm(EmbeddedParent()).apply {
+                childList.add(EmbeddedChild("child1"))
+            }
+        }
+        realm.query<EmbeddedChild>().find().single().run {
+            assertEquals("child1", id)
+        }
+    }
+
+    @Test
+    fun list_addWithIndex() {
+        realm.writeBlocking {
+            val element = EmbeddedChild("child1")
+            val parent = copyToRealm(EmbeddedParent()).apply {
+                childList.add(element)
+                childList.add(0, element)
```

Comment:
it would have been interesting to insert a different `EmbeddedChild` to assert that inserting at a specific index is honoured